shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/execute/PipelineJobExecutor.java

```diff
@@ -75,16 +74,16 @@ private Optional<JobConfigurationPOJO> getJobConfigPOJO(final DataChangedEvent e
     private void processEvent(final DataChangedEvent event, final JobConfigurationPOJO jobConfigPOJO) {
         boolean isDeleted = DataChangedEvent.Type.DELETED == event.getType();
         boolean isDisabled = jobConfigPOJO.isDisabled();
+        String jobId = jobConfigPOJO.getJobName();
+        JobType jobType = PipelineJobIdUtils.parseJobType(jobId);
         if (isDeleted || isDisabled) {
-            String jobId = jobConfigPOJO.getJobName();
             log.info("jobId={}, deleted={}, disabled={}", jobId, isDeleted, isDisabled);
-            // TODO refactor: dispatch to different job types
-            MigrationJobConfiguration jobConfig = YamlMigrationJobConfigurationSwapper.swapToObject(jobConfigPOJO.getJobParameter());
+            JobConfigEventProcess process = JobConfigEventProcessFactory.getInstance(jobType);
             if (isDeleted) {
-                new MigrationJobPreparer().cleanup(jobConfig);
-            } else if (PipelineJobProgressDetector.isJobSuccessful(jobConfig.getJobShardingCount(), MigrationJobAPIFactory.getInstance().getJobProgress(jobConfig).values())) {
+                process.cleanup(jobConfigPOJO.getJobParameter());
+            } else if (process.isJobSuccessful(jobConfigPOJO.getJobParameter())) {
                 log.info("isJobSuccessful=true");
-                new MigrationJobPreparer().cleanup(jobConfig);
+                process.cleanup(jobConfigPOJO.getJobParameter());
             }
             PipelineJobCenter.stop(jobId);
             return;
```

Comment:
After getting `jobType`, we could delegate all implementation to SPI and impl, so new type of job is free to implement it as needed.